ooni.go

```diff
@@ -140,11 +147,29 @@ func ParseConfig(b []byte) (*Config, error) {
 	return c, nil
 }
 
+func EnsureDefaultConfigDir() error {
+	home, err := GetOONIHome()
+	if err != nil {
+		return err
+	}
+
+	if _, e := os.Stat(filepath.Join(home, "db")); e != nil {
+		err = os.MkdirAll(filepath.Join(home, "db"), 0700)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 // ReadConfig reads the configuration from the path
 func ReadConfig(path string) (*Config, error) {
 	b, err := ioutil.ReadFile(path)
 
 	if os.IsNotExist(err) {
+		if err = EnsureDefaultConfigDir(); err != nil {
+			return nil, errors.Wrap(err, "Creating new config")
+		}
```

Comment:
I would not put this in here as this function is used to read the config file from a specific path. It doesn't make any assumption about what is on the disk. We should rather put this as something that is done when we construct the path to the config file inside of the ConfigDir

Reply:
See: https://github.com/OpenObservatory/gooni/commit/efb7e87d1b66a7a5dad48f460a8f8cb553a3761c